### PR TITLE
d2elklayout: Tweak elk layouts to prevent edge overlaps

### PR DESCRIPTION
**Pull request by tmc**
Before:
Screenshot 2023-04-09 at 7 40 57 PM

After:
Screenshot 2023-04-09 at 7 41 01 PM

Refs #1170 

<details></summary>Sample Graph</summary>

```
    	User <-> User : "spouse" {
		# o2o optional
		source-arrowhead: {
			shape: cf-one-required
		}
		target-arrowhead: {
			shape: cf-one
		}
		
	}
    	User <-> User : "children/parent" {
		# o2m optional
		source-arrowhead: {
			shape: cf-one-required
		}
		target-arrowhead: {
			shape: cf-many
		}
		
	}
    	User <-> Pet : "pets/owner" {
		# o2m optional
		source-arrowhead: {
			shape: cf-one-required
		}
		target-arrowhead: {
			shape: cf-many
		}
		
	}
    	User <-> Post : "posts/author" {
		# o2m optional
		source-arrowhead: {
			shape: cf-one-required
		}
		target-arrowhead: {
			shape: cf-many
		}
		
	}

# Tables
Pet: {
	shape: sql_table
	id: int {constraint: primary_key}
	owner_id: int {constraint: foreign_key}
}
Post: {
	shape: sql_table
	id: int {constraint: primary_key}
	text: string
	author_id: int {constraint: foreign_key}
}
User: {
	shape: sql_table
	id: int {constraint: primary_key}
	parent_id: int {constraint: foreign_key}
	spouse_id: int {constraint: foreign_key}
}
```

</details>

### CODE REVIEW

File: d2layouts/d2elklayout/layout.go [hunk 1]
```diff
@@ -96,7 +96,7 @@ var DefaultOpts = ConfigurableOpts{
 }
 
 var port_spacing = 40.
-var edge_node_spacing = 40
+var edge_node_spacing = 60
```
Comment (alixander) [inline d2layouts/d2elklayout/layout.go:99]: isn't this just arbitrary? what if a larger label comes along?

File: d2layouts/d2elklayout/layout.go [hunk 2]
```diff
@@ -306,7 +306,7 @@ func Layout(ctx context.Context, g *d2graph.Graph, opts *ConfigurableOpts) (err
 		if edge.Attributes.Label.Value != "" {
 			e.Labels = append(e.Labels, &ELKLabel{
 				Text:   edge.Attributes.Label.Value,
-				Width:  float64(edge.LabelDimensions.Width),
+				Width:  float64(edge.LabelDimensions.Width) * 1.6,
```
Comment (alixander) [inline d2layouts/d2elklayout/layout.go:309]: why this extra value?

**Review by alixander — CHANGES_REQUESTED** [review]
this doesn't seem robust. can you add a test to stable_test.go, and then sign your commit? and then regenerate tests? read contributing.md please

**Comment by tmc** [comment]
Both good points. This is the product of a minimal effort to hack in some improved elk rendering here.

I'll review the contribution guidelines and understand the testing strategy more and amend this.

File: d2layouts/d2elklayout/layout.go (hunk 1 above)
Comment (tmc) [reply d2layouts/d2elklayout/layout.go:99]: it is arbitrary, as the default is 10: https://www.eclipse.org/elk/reference/options/org-eclipse-elk-spacing-edgeNode.html -- is 40 not arbitary?

File: README.md
Comment (alixander) [inline README.md:1]: can you revert this plz? it's just pandoc version diff that causes this

File: d2layouts/d2elklayout/layout.go (hunk 1 above)
Comment (alixander) [reply d2layouts/d2elklayout/layout.go:99]: yeah it's the arbitrariness that caused the bug in the first place though, we're trying to fix it and not make it arbitrary

**Review by alixander — CHANGES_REQUESTED** [review]

File: d2layouts/d2elklayout/layout.go (hunk 1 above)
Comment (tmc) [reply d2layouts/d2elklayout/layout.go:99]: OK I'll poke around some more and see what I can suggest.

File: README.md
Comment (tmc) [reply README.md:1]: yeah this was a product of github's merge conflict UI, I'll improve and rebase this PR.

**Comment by tmc** [comment]
I don't really have the knowhow to contribute the right change here so I'm closing this.

**Pull request closed without merging**